src/Common/src/Interop/Windows/kernel32/Interop.CreateFile.cs

```diff
@@ -15,38 +15,40 @@ internal partial class Kernel32
         /// WARNING: The private methods do not implicitly handle long paths. Use CreateFile.
         /// </summary>
         [DllImport(Libraries.Kernel32, EntryPoint = "CreateFileW", SetLastError = true, CharSet = CharSet.Unicode, BestFitMapping = false, ExactSpelling = true)]
-        private static extern SafeFileHandle CreateFilePrivate(
+        private unsafe static extern SafeFileHandle CreateFilePrivate(
             string lpFileName,
             int dwDesiredAccess,
             FileShare dwShareMode,
-            [In] ref SECURITY_ATTRIBUTES securityAttrs,
+            SECURITY_ATTRIBUTES* securityAttrs,
             FileMode dwCreationDisposition,
             int dwFlagsAndAttributes,
             IntPtr hTemplateFile);
 
-        internal static SafeFileHandle CreateFile(
+        internal unsafe static SafeFileHandle CreateFile(
             string lpFileName,
             int dwDesiredAccess,
             FileShare dwShareMode,
-            [In] ref SECURITY_ATTRIBUTES securityAttrs,
+            ref SECURITY_ATTRIBUTES securityAttrs,
             FileMode dwCreationDisposition,
             int dwFlagsAndAttributes,
             IntPtr hTemplateFile)
         {
             lpFileName = PathInternal.EnsureExtendedPrefixOverMaxPath(lpFileName);
-            return CreateFilePrivate(lpFileName, dwDesiredAccess, dwShareMode, ref securityAttrs, dwCreationDisposition, dwFlagsAndAttributes, hTemplateFile);
+            fixed (SECURITY_ATTRIBUTES* sa = &securityAttrs)
```

Comment:
what is the advantage of this over ref?

Reply:
Can pass null, which is the normal case.